Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,136 @@
+
+        log(level: .verbose, message: message, file: file, function: function, line: line, completion: completion)
+    }
+
+    public class func debug(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line,
+                            completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
+
+        log(level: .debug, message: message, file: file, function: function, line: line, completion: completion)
+    }
+
+    public class func info(_ message: @autoclosure () -> String,
+                           file: String = #file,
+                           function: String = #function,
+                           line: Int = #line,
+                           completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
+
+        log(level: .info, message: message, file: file, function: function, line: line, completion: completion)
+    }
+
+    public class func warning(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line,
+                              completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
+
+        log(level: .warning, message: message, file: file, function: function, line: line, completion: completion)
+    }
+
+    public class func error(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line,
+                            completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
+
+        log(level: .error, message: message, file: file, function: function, line: line, completion: completion)
+    }
+
+    public class func log(level: Level,
+                          message: @autoclosure () -> String,
+                          file: String = #file,
+                          function: String = #function,
+                          line: Int = #line,
+                          completion: @escaping (LogDestination, Item, Error?) -> Void = Log.defaultLogItemWrittenHandler) {
+
+        let item = Log.Item(level: level, message: message(), file: file,
+                            thread: threadName(), function: function, line: line)
+
+        for destination in destinations {
+            if itemShouldBeLogged(destination: destination, item: item) {
+                destination.write(item: item, completion: completion)
+            }
+        }
+    }
+
+    // MARK:- Private Methods
+
+    private class func threadName() -> String {
+
+        if Thread.isMainThread {
```

Comment:
Just a thought. Perhaps might be worth changing this one to a `guard`. This will not only return early if we are on the main thread, but also will remove 1 level of indentation.